### PR TITLE
New module: scvitools/solo single-cell doublet detection

### PR DESCRIPTION
**Pull request by LeonHafner**
## PR checklist
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`

### CODE REVIEW

**Comment by LeonHafner** [comment]
@nf-core-bot fix linting

**Comment by LeonHafner** [comment]
Module is producing inconsistent hashes for the output anndata object across two nf-test runs.

File: modules/nf-core/scvitools/solo/templates/solo.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import anndata as ad
+import scvi
+import torch
+from scvi.external import SOLO
+from scvi.model import SCVI
+from threadpoolctl import threadpool_limits
+
+torch.set_float32_matmul_precision("medium")
+scvi.settings.seed = 0
```
Comment (nictru) [inline modules/nf-core/scvitools/solo/templates/solo.py:13]:
```suggestion
scvi.settings.seed = 0
torch.manual_seed(0)
np.random.seed(0)
random.seed(0)
torch.use_deterministic_algorithms(True)
```

File: modules/nf-core/scvitools/solo/templates/solo.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+import platform
```
Comment (nictru) [inline modules/nf-core/scvitools/solo/templates/solo.py:3]:
```suggestion
import platform
import random
```

File: modules/nf-core/scvitools/solo/templates/solo.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+import anndata as ad
```
Comment (nictru) [inline modules/nf-core/scvitools/solo/templates/solo.py:5]:
```suggestion
import numpy as np
import anndata as ad
```

**Review by nictru — CHANGES_REQUESTED** [review]
Ideas for removing sources of randomness

**Comment by nictru** [comment]
Things left to be done:

- [ ] Remove unnecessary seeds
- [ ] Update scvi-tools version and remove batch stuff

**Comment by nictru** [comment]
Should switch this to an R implementation. [This Vignette](https://cellgeni.github.io/notebooks/html/new-10kPBMC-SoupX.html) can serve as a template. Should use AnndataR if the file ends with `.h5ad` otherwise `readRDS`.

**Review by nictru — APPROVED** [review]
